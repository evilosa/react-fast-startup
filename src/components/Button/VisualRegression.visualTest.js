import { Chrome } from 'navalia';
import { toMatchImageSnapshot } from 'jest-image-snapshot';

expect.extend({ toMatchImageSnapshot });

describe('Visual Regressions', () => {
  let chrome = null;
  let server = null;

  beforeEach(() => {
    server = require('../../../scripts/server');
    chrome = new Chrome({
      flags: {
        headless: true,
        disableGpu: true,
        hideScrollbars: true,
        noSandbox: true
      }
    });
  });

  afterEach(() => {
    chrome.done();
    server.close();
  });

  it('should NEVER happen', () => {
    return chrome.goto('http://localhost:3000/')
      .then(() => chrome.screenshot())
      .then((image) => {
        console.log(`Image length: ${image.length}`);
        expect(image).toMatchImageSnapshot();
      });
  });
});