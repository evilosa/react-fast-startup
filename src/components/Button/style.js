import Color from 'color';

export const ButtonColors = {
  baseColor: '#fff',
  baseBackground: '#222222',
  primaryBackground: '#0074d9',
  warningBackground: '#f00',
};

export const createButtonStyle = (colors = ButtonColors) => ({
  base: {
    color: colors.baseColor,
    ':hover': {
      background: Color(colors.baseBackground).lighten(0.2).hex(),
    },
    background: colors.baseBackground,
  },

  primary: {
    background: colors.baseBackground,
  },

  small: {
    fontSize: '12px',
    padding: '5px',
    fontWeight: '300',
    borderRadius: '4px',
  },

  normal: {
    fontSize: '16px',
    fontWeight: '300',
    padding: '10px',
    borderRadius: '4px',
  },

  big: {
    fontSize: '24px',
    fontWeight: '300',
    padding: '15px',
    minWidth: '150px',
    borderRadius: '4px',
  },

  warning: {
    background: colors.warningBackground,
  }
});

export const ButtonStyle = createButtonStyle(ButtonColors);

export default ButtonStyle;
