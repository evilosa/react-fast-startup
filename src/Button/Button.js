// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';
import style from './style';

class Button extends React.Component {

  static propTypes = {
    /** Default style for button */
    style: PropTypes.object,
  };

  static defaultProps = {
    style,
  };

  _onClick = () => {
    const { style } = this.props;
    console.log('Button style');
    console.log(style);
  };

  render() {
    const { style, inlineStyle, children } = this.props;

    return (
      <button style={[style.base, style.primary, inlineStyle]} onClick={this._onClick} type="Button">{children}</button>
    )
  }
}

//$FlowFixMe
export default Radium(Button);