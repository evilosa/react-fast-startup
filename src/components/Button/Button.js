// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';
import style from './style';

class Button extends React.Component {

  static propTypes = {
    /** Default style for button */
    style: PropTypes.object,
    type: PropTypes.string,
    onClick: PropTypes.func
  };

  static defaultProps = {
    style,
    type: 'normal',
  };

  _onClick = () => {};

  render() {
    const { style, onClick, inlineStyle, children, type } = this.props;

    return (
      <button style={[style.base, style.primary, style[type], inlineStyle]} onClick={onClick} type="Button">{children}</button>
    )
  }
}

//$FlowFixMe
export default Radium(Button);