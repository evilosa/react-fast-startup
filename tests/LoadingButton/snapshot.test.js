import React from 'react';
import Button from '../../src/Button';

const button = shallow(<Button />);

describe('Button component', () => {

  it('should render without throwing an error', () => {
    expect(shallow(<Button/>).exists(<button type="Button333">344</button>)).toBe(true);
  });

  it('renders correctly', () => {
    expect(button).toMatchSnapshot();
  });

  it('should render to static HTML', function() {
    expect(render(<Button>Her</Button>).text()).toEqual('Her');
  });

  test('render a label', () => {
    const wrapper = shallow(
      <div>Hello Jest!</div>
    );
    expect(wrapper).toMatchSnapshot();
  });
});