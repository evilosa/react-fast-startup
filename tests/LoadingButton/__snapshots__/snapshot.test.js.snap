// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a label 1`] = `
<div>
  Hello Jest!
</div>
`;

exports[`renders correctly 1`] = `
<button
  data-radium={true}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "#0074d9",
      "color": "#fff",
    }
  }
  type="Button"
/>
`;
